tests/test_split_collect.py

```diff
@@ -0,0 +1,120 @@
+import glob
+import os
+import random
+import shutil
+import string
+import subprocess
+
+import pytest
+import sh
+
+from pipeline.translate.splitter import split_file
+
+COMPRESSION_CMD = "zstdmt"
+
+OUTPUT_DIR = "tests_data"
+
+
+@pytest.fixture(scope="function")
+def clean():
+    if os.path.exists(OUTPUT_DIR):
+        shutil.rmtree(OUTPUT_DIR)
+    os.makedirs(OUTPUT_DIR, exist_ok=True)
+
+
+def generate_dataset(length, path):
+    words = [
+        "".join([random.choice(string.ascii_letters) for _ in range(random.randint(1, 10))])
+        for _ in range(20)
+    ]
+    sentences = []
+    for i in range(length):
+        sentence = " ".join([words[random.randint(0, 19)] for _ in range(random.randint(1, 200))])
+        sentences.append(sentence)
+
+    with open(path, "w") as f:
+        f.write("\n".join(sentences))
+
+    sh.zstdmt(path)
+
+
+def decompress(path):
+    sh.zstdmt("-d", path)
+
+
+def imitate_translate(dir, suffix):
+    for file in glob.glob(f"{dir}/file.?.zst") + glob.glob(f"{dir}/file.??.zst"):
+        print(file)
+        decompress(file)
+        shutil.copy(file[:-4], file[:-4] + suffix)
+
+
+def read_file(path):
+    with open(path) as f:
+        return f.read()
+
+
+def test_split_collect_mono(clean):
+    os.environ["COMPRESSION_CMD"] = COMPRESSION_CMD
+    length = 1234
+    path = os.path.join(OUTPUT_DIR, "mono.in")
+    output = os.path.join(OUTPUT_DIR, "mono.output")
+    output_compressed = f"{output}.zst"
+    generate_dataset(length, path)
+
+    split_file(
+        mono_path=f"{path}.zst",
+        output_dir=OUTPUT_DIR,
+        num_parts=10,
+        compression_cmd=COMPRESSION_CMD,
+    )
+
+    expected_files = set([f"{OUTPUT_DIR}/file.{i}.zst" for i in range(1, 11)])
+    assert set(glob.glob(f"{OUTPUT_DIR}/file.*.zst")) == expected_files
```

Comment:
Suggestion: I would prefer asserting on a static list of values rather than computed values. This makes it easier to tell just from the code how something is working, and that it is working correctly. Dynamically generated expectations are harder to reason with.

```suggestion
    assert set(glob.glob(f"{OUTPUT_DIR}/file.*.zst")) == set(
        [
            "tests_data/file.1.zst",
            "tests_data/file.2.zst",
            "tests_data/file.3.zst",
            "tests_data/file.4.zst",
            "tests_data/file.5.zst",
            "tests_data/file.6.zst",
            "tests_data/file.7.zst",
            "tests_data/file.8.zst",
            "tests_data/file.9.zst",
            "tests_data/file.10.zst",
        ]
    )
```

Reply:
I prefer having shorter unit tests. I added comments to clarify what files are generated.